pkg/virt-handler/vm.go

```diff
@@ -408,115 +511,178 @@ func (d *VMHandlerDispatch) injectDiskAuth(vm *v1.VirtualMachine) (*v1.VirtualMa
+func (d *VirtualMachineController) processVmUpdate(vm *v1.VirtualMachine, shouldDeleteVM bool) error {
 
-	if shouldDeleteVm {
+	if shouldDeleteVM || vm.ObjectMeta.DeletionTimestamp != nil {
 		// Since the VM was not in the cache, we delete it
 		err := d.domainManager.KillVM(vm)
 		if err != nil {
-			return false, err
+			return err
 		}
 
 		// remove any defined libvirt secrets associated with this vm
 		err = d.domainManager.RemoveVMSecrets(vm)
 		if err != nil {
-			return false, err
+			return err
 		}
 
 		err = registrydisk.CleanupEphemeralDisks(vm)
 		if err != nil {
-			return false, err
+			return err
 		}
 
 		err = watchdog.WatchdogFileRemove(d.virtShareDir, vm)
 		if err != nil {
-			return false, err
+			return err
 		}
 
-		return false, d.configDisk.Undefine(vm)
+		return d.configDisk.Undefine(vm)
 	} else if isWorthSyncing(vm) == false {
 		// nothing to do here.
-		return false, nil
+		return nil
 	}
 
 	hasWatchdog, err := watchdog.WatchdogFileExists(d.virtShareDir, vm)
 	if err != nil {
 		log.Log.Object(vm).Reason(err).V(3).Error("Error accessing virt-launcher watchdog file.")
-		return false, err
+		return err
 	}
 	if hasWatchdog == false {
 		log.Log.Object(vm).Reason(err).V(3).Error("Could not detect virt-launcher watchdog file.")
-		return false, goerror.New(fmt.Sprintf("No watchdog file found for vm"))
+		return goerror.New(fmt.Sprintf("No watchdog file found for vm"))
 	}
 
 	isPending, err := d.configDisk.Define(vm)
-	if err != nil || isPending == true {
-		return isPending, err
+	if err != nil {
+		return err
+	}
+
+	if isPending {
+		log.Log.Object(vm).V(3).Info("Synchronizing is in a pending state.")
+		d.Queue.AddAfter(controller.VirtualMachineKey(vm), 1*time.Second)
+		return nil
 	}
 
 	// Synchronize the VM state
 	vm, err = MapPersistentVolumes(vm, d.clientset.CoreV1().RESTClient(), vm.ObjectMeta.Namespace)
 	if err != nil {
-		return false, err
+		return err
 	}
 
 	// Map Container Registry Disks to block devices Libvirt can consume
 	vm, err = registrydisk.MapRegistryDisks(vm)
 	if err != nil {
-		return false, err
+		return err
 	}
 
 	vm, err = d.injectDiskAuth(vm)
 	if err != nil {
-		return false, err
+		return err
 	}
 
 	// Map whatever devices are being used for config-init
 	vm, err = cloudinit.MapCloudInitDisks(vm)
 	if err != nil {
-		return false, err
+		return err
 	}
 
 	// TODO MigrationNodeName should be a pointer
 	if vm.Status.MigrationNodeName != "" {
 		// Only sync if the VM is not marked as migrating.
 		// Everything except shutting down the VM is not
 		// permitted when it is migrating.
-		return false, nil
+		return nil
 	}
 
 	// TODO check if found VM has the same UID like the domain,
 	// if not, delete the Domain first
-	newCfg, err := d.domainManager.SyncVM(vm)
-	if err != nil {
-		return false, err
-	}
-
-	return false, d.updateVMStatus(vm, newCfg)
+	_, err = d.domainManager.SyncVM(vm)
```

Comment:
Okay, so we get the domain status updates from the domain informer now and don't have to do anything with the domain object here. I that makes sense.